{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T23:26:25.728262Z",
     "iopub.status.busy": "2023-11-08T23:26:25.728076Z",
     "iopub.status.idle": "2023-11-08T23:26:25.815213Z",
     "shell.execute_reply": "2023-11-08T23:26:25.814640Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-08T23:26:25.817488Z",
     "iopub.status.busy": "2023-11-08T23:26:25.817306Z",
     "iopub.status.idle": "2023-11-09T01:15:04.152720Z",
     "shell.execute_reply": "2023-11-09T01:15:04.152198Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:26:26.002444: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35223\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:27:49.250292: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.261819: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.262051: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.265227: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.265415: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.265586: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.341598: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.341810: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.341978: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:49.342134: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 18:27:49.342534: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:27:50.274573: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:50.274901: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:50.275088: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:50.275303: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:50.275480: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 18:27:50.275627: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 18:27:50.275656: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 18:27:50.298415: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 18:27:50.512734: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21/lstm_cell_21/recurrent_kernel/Assign' id:3501 op device:{requested: '', assigned: ''} def:{{{node lstm_21/lstm_cell_21/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21/lstm_cell_21/recurrent_kernel, lstm_21/lstm_cell_21/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:27:50.643669: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:27:50.676686: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31685, 95)\n",
      "Train on 31685 samples, validate on 3538 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:27:55.061687: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/conv2d_1/bias/v/Assign' id:17012 op device:{requested: '', assigned: ''} def:{{{node training/Adam/conv2d_1/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/conv2d_1/bias/v, training/Adam/conv2d_1/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:27:58.878528: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 18:28:01.045792: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "2023-11-08 18:28:01.053975: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31685/31685 [==============================] - ETA: 0s - loss: 3.9982"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 18:28:21.783667: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 2.04270, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 28s 885us/sample - loss: 3.9982 - val_loss: 2.0427\n",
      "Epoch 2/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.9587\n",
      "Epoch 2: val_loss improved from 2.04270 to 1.60937, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 20s 637us/sample - loss: 1.9587 - val_loss: 1.6094\n",
      "Epoch 3/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.6181\n",
      "Epoch 3: val_loss improved from 1.60937 to 1.49633, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 20s 636us/sample - loss: 1.6181 - val_loss: 1.4963\n",
      "Epoch 4/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.5474\n",
      "Epoch 4: val_loss improved from 1.49633 to 1.46315, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 596us/sample - loss: 1.5474 - val_loss: 1.4632\n",
      "Epoch 5/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.5234\n",
      "Epoch 5: val_loss improved from 1.46315 to 1.45242, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.5234 - val_loss: 1.4524\n",
      "Epoch 6/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.5054\n",
      "Epoch 6: val_loss improved from 1.45242 to 1.44457, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 694us/sample - loss: 1.5054 - val_loss: 1.4446\n",
      "Epoch 7/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4960\n",
      "Epoch 7: val_loss improved from 1.44457 to 1.43874, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 694us/sample - loss: 1.4960 - val_loss: 1.4387\n",
      "Epoch 8/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4929\n",
      "Epoch 8: val_loss improved from 1.43874 to 1.43543, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 20s 624us/sample - loss: 1.4929 - val_loss: 1.4354\n",
      "Epoch 9/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.5092\n",
      "Epoch 9: val_loss improved from 1.43543 to 1.43429, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 20s 630us/sample - loss: 1.5092 - val_loss: 1.4343\n",
      "Epoch 10/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4857\n",
      "Epoch 10: val_loss improved from 1.43429 to 1.42538, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 693us/sample - loss: 1.4857 - val_loss: 1.4254\n",
      "Epoch 11/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4884\n",
      "Epoch 11: val_loss improved from 1.42538 to 1.42433, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 20s 643us/sample - loss: 1.4884 - val_loss: 1.4243\n",
      "Epoch 12/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4736\n",
      "Epoch 12: val_loss improved from 1.42433 to 1.41642, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 609us/sample - loss: 1.4736 - val_loss: 1.4164\n",
      "Epoch 13/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4697\n",
      "Epoch 13: val_loss improved from 1.41642 to 1.41309, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 607us/sample - loss: 1.4697 - val_loss: 1.4131\n",
      "Epoch 14/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4617\n",
      "Epoch 14: val_loss did not improve from 1.41309\n",
      "31685/31685 [==============================] - 19s 602us/sample - loss: 1.4617 - val_loss: 1.4147\n",
      "Epoch 15/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4617\n",
      "Epoch 15: val_loss improved from 1.41309 to 1.41179, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 606us/sample - loss: 1.4617 - val_loss: 1.4118\n",
      "Epoch 16/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.5363\n",
      "Epoch 16: val_loss did not improve from 1.41179\n",
      "31685/31685 [==============================] - 21s 674us/sample - loss: 1.5363 - val_loss: 1.4186\n",
      "Epoch 17/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4844\n",
      "Epoch 17: val_loss did not improve from 1.41179\n",
      "31685/31685 [==============================] - 20s 631us/sample - loss: 1.4844 - val_loss: 1.4147\n",
      "Epoch 18/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4629\n",
      "Epoch 18: val_loss did not improve from 1.41179\n",
      "31685/31685 [==============================] - 20s 631us/sample - loss: 1.4629 - val_loss: 1.4136\n",
      "Epoch 19/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4614\n",
      "Epoch 19: val_loss improved from 1.41179 to 1.40365, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 20s 618us/sample - loss: 1.4614 - val_loss: 1.4037\n",
      "Epoch 20/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4653\n",
      "Epoch 20: val_loss did not improve from 1.40365\n",
      "31685/31685 [==============================] - 20s 635us/sample - loss: 1.4653 - val_loss: 1.4073\n",
      "Epoch 21/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4616\n",
      "Epoch 21: val_loss did not improve from 1.40365\n",
      "31685/31685 [==============================] - 19s 594us/sample - loss: 1.4616 - val_loss: 1.4151\n",
      "Epoch 22/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4577\n",
      "Epoch 22: val_loss did not improve from 1.40365\n",
      "31685/31685 [==============================] - 21s 659us/sample - loss: 1.4577 - val_loss: 1.4138\n",
      "Epoch 23/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4491\n",
      "Epoch 23: val_loss did not improve from 1.40365\n",
      "31685/31685 [==============================] - 22s 685us/sample - loss: 1.4491 - val_loss: 1.4038\n",
      "Epoch 24/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4414\n",
      "Epoch 24: val_loss improved from 1.40365 to 1.40116, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 696us/sample - loss: 1.4414 - val_loss: 1.4012\n",
      "Epoch 25/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4410\n",
      "Epoch 25: val_loss did not improve from 1.40116\n",
      "31685/31685 [==============================] - 22s 692us/sample - loss: 1.4410 - val_loss: 1.4017\n",
      "Epoch 26/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4362\n",
      "Epoch 26: val_loss improved from 1.40116 to 1.39699, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 21s 664us/sample - loss: 1.4362 - val_loss: 1.3970\n",
      "Epoch 27/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4370\n",
      "Epoch 27: val_loss did not improve from 1.39699\n",
      "31685/31685 [==============================] - 20s 644us/sample - loss: 1.4370 - val_loss: 1.3971\n",
      "Epoch 28/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4317\n",
      "Epoch 28: val_loss improved from 1.39699 to 1.39230, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 21s 676us/sample - loss: 1.4317 - val_loss: 1.3923\n",
      "Epoch 29/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4293\n",
      "Epoch 29: val_loss did not improve from 1.39230\n",
      "31685/31685 [==============================] - 20s 638us/sample - loss: 1.4293 - val_loss: 1.3945\n",
      "Epoch 30/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4278\n",
      "Epoch 30: val_loss improved from 1.39230 to 1.38861, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 602us/sample - loss: 1.4278 - val_loss: 1.3886\n",
      "Epoch 31/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4264\n",
      "Epoch 31: val_loss did not improve from 1.38861\n",
      "31685/31685 [==============================] - 19s 600us/sample - loss: 1.4264 - val_loss: 1.3916\n",
      "Epoch 32/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4217\n",
      "Epoch 32: val_loss improved from 1.38861 to 1.38628, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 607us/sample - loss: 1.4217 - val_loss: 1.3863\n",
      "Epoch 33/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4229\n",
      "Epoch 33: val_loss did not improve from 1.38628\n",
      "31685/31685 [==============================] - 21s 670us/sample - loss: 1.4229 - val_loss: 1.3867\n",
      "Epoch 34/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4176\n",
      "Epoch 34: val_loss improved from 1.38628 to 1.38078, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 21s 656us/sample - loss: 1.4176 - val_loss: 1.3808\n",
      "Epoch 35/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4179\n",
      "Epoch 35: val_loss improved from 1.38078 to 1.37964, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 698us/sample - loss: 1.4179 - val_loss: 1.3796\n",
      "Epoch 36/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4148\n",
      "Epoch 36: val_loss improved from 1.37964 to 1.37819, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 21s 671us/sample - loss: 1.4148 - val_loss: 1.3782\n",
      "Epoch 37/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4129\n",
      "Epoch 37: val_loss improved from 1.37819 to 1.37777, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 608us/sample - loss: 1.4129 - val_loss: 1.3778\n",
      "Epoch 38/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4122\n",
      "Epoch 38: val_loss did not improve from 1.37777\n",
      "31685/31685 [==============================] - 19s 603us/sample - loss: 1.4122 - val_loss: 1.3840\n",
      "Epoch 39/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4105\n",
      "Epoch 39: val_loss did not improve from 1.37777\n",
      "31685/31685 [==============================] - 22s 683us/sample - loss: 1.4105 - val_loss: 1.3800\n",
      "Epoch 40/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4108\n",
      "Epoch 40: val_loss improved from 1.37777 to 1.37507, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 692us/sample - loss: 1.4108 - val_loss: 1.3751\n",
      "Epoch 41/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4087\n",
      "Epoch 41: val_loss improved from 1.37507 to 1.37228, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 694us/sample - loss: 1.4087 - val_loss: 1.3723\n",
      "Epoch 42/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4109\n",
      "Epoch 42: val_loss did not improve from 1.37228\n",
      "31685/31685 [==============================] - 20s 623us/sample - loss: 1.4109 - val_loss: 1.3748\n",
      "Epoch 43/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4057\n",
      "Epoch 43: val_loss improved from 1.37228 to 1.36766, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.4057 - val_loss: 1.3677\n",
      "Epoch 44/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4047\n",
      "Epoch 44: val_loss did not improve from 1.36766\n",
      "31685/31685 [==============================] - 19s 611us/sample - loss: 1.4047 - val_loss: 1.3714\n",
      "Epoch 45/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4028\n",
      "Epoch 45: val_loss did not improve from 1.36766\n",
      "31685/31685 [==============================] - 19s 598us/sample - loss: 1.4028 - val_loss: 1.3746\n",
      "Epoch 46/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4020\n",
      "Epoch 46: val_loss did not improve from 1.36766\n",
      "31685/31685 [==============================] - 19s 595us/sample - loss: 1.4020 - val_loss: 1.3721\n",
      "Epoch 47/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4025\n",
      "Epoch 47: val_loss did not improve from 1.36766\n",
      "31685/31685 [==============================] - 19s 598us/sample - loss: 1.4025 - val_loss: 1.3691\n",
      "Epoch 48/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4021\n",
      "Epoch 48: val_loss did not improve from 1.36766\n",
      "31685/31685 [==============================] - 19s 606us/sample - loss: 1.4021 - val_loss: 1.3697\n",
      "Epoch 49/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3983\n",
      "Epoch 49: val_loss did not improve from 1.36766\n",
      "31685/31685 [==============================] - 19s 599us/sample - loss: 1.3983 - val_loss: 1.3704\n",
      "Epoch 50/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3976\n",
      "Epoch 50: val_loss improved from 1.36766 to 1.36537, saving model to ./checkpoints/unknown_person_baseline_p13_20.h5\n",
      "31685/31685 [==============================] - 21s 648us/sample - loss: 1.3976 - val_loss: 1.3654\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:44:59.232351: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_1_1/kernel/Assign' id:24601 op device:{requested: '', assigned: ''} def:{{{node dense_1_1/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_1_1/kernel, dense_1_1/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:45:00.940416: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_2_1/lstm_cell_39/kernel/v/Assign' id:25925 op device:{requested: '', assigned: ''} def:{{{node lstm_2_1/lstm_cell_39/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_2_1/lstm_cell_39/kernel/v, lstm_2_1/lstm_cell_39/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 18:45:02.291203: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1778)\n",
      "(1514, 1778)\n",
      "(1644, 1778)\n",
      "(1764, 1778)\n",
      "(1836, 1778)\n",
      "(1699, 1778)\n",
      "(1369, 1778)\n",
      "(1778, 1778)\n",
      "(1631, 1778)\n",
      "(1692, 1778)\n",
      "(1920, 1778)\n",
      "(1679, 1778)\n",
      "(1800, 1778)\n",
      "(1860, 1778)\n",
      "(1692, 1778)\n",
      "(1824, 1778)\n",
      "(958, 1778)\n",
      "(1680, 1778)\n",
      "(1860, 1778)\n",
      "{1: 8.835532591878277, 2: 4.713636136964317, 4: 7.90378030686214, 5: 8.155287190244554, 6: 6.608267960035775, 8: 8.871468530472171, 9: 4.448400699186118, 10: 7.840086109472236, 11: 5.680803662502162, 12: 9.19924213820389, 17: 8.416906885066263, 19: 7.720030231574639, 21: 10.0, 22: 1.0, 25: 7.580009079834542, 26: 6.915703256500144, 27: 2.356488422563324, 28: 6.539026773854481, 29: 3.342115471921486}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2793075/631073528.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31685 samples, validate on 3538 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:48:36.318374: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31685/31685 [==============================] - ETA: 0s - loss: 9.9488\n",
      "Epoch 1: val_loss improved from inf to 1.40333, saving model to ./checkpoints/unknown_person_p13_20.h5\n",
      "31685/31685 [==============================] - 25s 793us/sample - loss: 9.9488 - val_loss: 1.4033\n",
      "Epoch 2/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.8996\n",
      "Epoch 2: val_loss improved from 1.40333 to 1.36619, saving model to ./checkpoints/unknown_person_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 614us/sample - loss: 9.8996 - val_loss: 1.3662\n",
      "Epoch 3/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.8323\n",
      "Epoch 3: val_loss improved from 1.36619 to 1.36527, saving model to ./checkpoints/unknown_person_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 603us/sample - loss: 9.8323 - val_loss: 1.3653\n",
      "Epoch 4/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.8142\n",
      "Epoch 4: val_loss did not improve from 1.36527\n",
      "31685/31685 [==============================] - 19s 600us/sample - loss: 9.8142 - val_loss: 1.3822\n",
      "Epoch 5/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.7671\n",
      "Epoch 5: val_loss did not improve from 1.36527\n",
      "31685/31685 [==============================] - 20s 643us/sample - loss: 9.7671 - val_loss: 1.3704\n",
      "Epoch 6/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.7355\n",
      "Epoch 6: val_loss improved from 1.36527 to 1.36199, saving model to ./checkpoints/unknown_person_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 693us/sample - loss: 9.7355 - val_loss: 1.3620\n",
      "Epoch 7/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.7247\n",
      "Epoch 7: val_loss did not improve from 1.36199\n",
      "31685/31685 [==============================] - 22s 687us/sample - loss: 9.7247 - val_loss: 1.3711\n",
      "Epoch 8/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.7015\n",
      "Epoch 8: val_loss improved from 1.36199 to 1.35774, saving model to ./checkpoints/unknown_person_p13_20.h5\n",
      "31685/31685 [==============================] - 21s 676us/sample - loss: 9.7015 - val_loss: 1.3577\n",
      "Epoch 9/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6773\n",
      "Epoch 9: val_loss did not improve from 1.35774\n",
      "31685/31685 [==============================] - 20s 619us/sample - loss: 9.6773 - val_loss: 1.3589\n",
      "Epoch 10/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6849\n",
      "Epoch 10: val_loss did not improve from 1.35774\n",
      "31685/31685 [==============================] - 21s 664us/sample - loss: 9.6849 - val_loss: 1.3618\n",
      "Epoch 11/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6784\n",
      "Epoch 11: val_loss improved from 1.35774 to 1.35044, saving model to ./checkpoints/unknown_person_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 688us/sample - loss: 9.6784 - val_loss: 1.3504\n",
      "Epoch 12/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6642\n",
      "Epoch 12: val_loss did not improve from 1.35044\n",
      "31685/31685 [==============================] - 22s 692us/sample - loss: 9.6642 - val_loss: 1.3510\n",
      "Epoch 13/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6411\n",
      "Epoch 13: val_loss improved from 1.35044 to 1.34950, saving model to ./checkpoints/unknown_person_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 689us/sample - loss: 9.6411 - val_loss: 1.3495\n",
      "Epoch 14/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6297\n",
      "Epoch 14: val_loss did not improve from 1.34950\n",
      "31685/31685 [==============================] - 20s 637us/sample - loss: 9.6297 - val_loss: 1.3593\n",
      "Epoch 15/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6326\n",
      "Epoch 15: val_loss did not improve from 1.34950\n",
      "31685/31685 [==============================] - 19s 603us/sample - loss: 9.6326 - val_loss: 1.3530\n",
      "Epoch 16/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6230\n",
      "Epoch 16: val_loss did not improve from 1.34950\n",
      "31685/31685 [==============================] - 20s 623us/sample - loss: 9.6230 - val_loss: 1.3565\n",
      "Epoch 17/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6061\n",
      "Epoch 17: val_loss did not improve from 1.34950\n",
      "31685/31685 [==============================] - 19s 610us/sample - loss: 9.6061 - val_loss: 1.3549\n",
      "Epoch 18/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.5868\n",
      "Epoch 18: val_loss improved from 1.34950 to 1.34800, saving model to ./checkpoints/unknown_person_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 615us/sample - loss: 9.5868 - val_loss: 1.3480\n",
      "Epoch 19/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.5676\n",
      "Epoch 19: val_loss did not improve from 1.34800\n",
      "31685/31685 [==============================] - 20s 630us/sample - loss: 9.5676 - val_loss: 1.3554\n",
      "Epoch 20/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.5706\n",
      "Epoch 20: val_loss did not improve from 1.34800\n",
      "31685/31685 [==============================] - 19s 610us/sample - loss: 9.5706 - val_loss: 1.3597\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:55:29.607734: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_27_2/lstm_cell_101/bias/Assign' id:42389 op device:{requested: '', assigned: ''} def:{{{node lstm_27_2/lstm_cell_101/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_27_2/lstm_cell_101/bias, lstm_27_2/lstm_cell_101/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 18:55:32.003281: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_2_2/kernel/m/Assign' id:45231 op device:{requested: '', assigned: ''} def:{{{node dense_2_2/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_2_2/kernel/m, dense_2_2/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31685 samples, validate on 3538 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:55:37.070078: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:55:49.570039: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3951"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 18:56:10.162997: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.36758, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 27s 844us/sample - loss: 1.3951 - val_loss: 1.3676\n",
      "Epoch 2/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3964\n",
      "Epoch 2: val_loss improved from 1.36758 to 1.36519, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 609us/sample - loss: 1.3964 - val_loss: 1.3652\n",
      "Epoch 3/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3934\n",
      "Epoch 3: val_loss improved from 1.36519 to 1.36207, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 609us/sample - loss: 1.3934 - val_loss: 1.3621\n",
      "Epoch 4/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3912\n",
      "Epoch 4: val_loss improved from 1.36207 to 1.36140, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 615us/sample - loss: 1.3912 - val_loss: 1.3614\n",
      "Epoch 5/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3938\n",
      "Epoch 5: val_loss did not improve from 1.36140\n",
      "31685/31685 [==============================] - 20s 617us/sample - loss: 1.3938 - val_loss: 1.3629\n",
      "Epoch 6/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3868\n",
      "Epoch 6: val_loss did not improve from 1.36140\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.3868 - val_loss: 1.3689\n",
      "Epoch 7/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3907\n",
      "Epoch 7: val_loss did not improve from 1.36140\n",
      "31685/31685 [==============================] - 19s 601us/sample - loss: 1.3907 - val_loss: 1.3623\n",
      "Epoch 8/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3888\n",
      "Epoch 8: val_loss did not improve from 1.36140\n",
      "31685/31685 [==============================] - 19s 606us/sample - loss: 1.3888 - val_loss: 1.3697\n",
      "Epoch 9/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3881\n",
      "Epoch 9: val_loss improved from 1.36140 to 1.36085, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 20s 631us/sample - loss: 1.3881 - val_loss: 1.3609\n",
      "Epoch 10/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3836\n",
      "Epoch 10: val_loss improved from 1.36085 to 1.35989, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 22s 689us/sample - loss: 1.3836 - val_loss: 1.3599\n",
      "Epoch 11/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3834\n",
      "Epoch 11: val_loss did not improve from 1.35989\n",
      "31685/31685 [==============================] - 20s 623us/sample - loss: 1.3834 - val_loss: 1.3639\n",
      "Epoch 12/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3823\n",
      "Epoch 12: val_loss improved from 1.35989 to 1.35889, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 21s 656us/sample - loss: 1.3823 - val_loss: 1.3589\n",
      "Epoch 13/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3822\n",
      "Epoch 13: val_loss did not improve from 1.35889\n",
      "31685/31685 [==============================] - 19s 590us/sample - loss: 1.3822 - val_loss: 1.3598\n",
      "Epoch 14/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3836\n",
      "Epoch 14: val_loss improved from 1.35889 to 1.35700, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 595us/sample - loss: 1.3836 - val_loss: 1.3570\n",
      "Epoch 15/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3791\n",
      "Epoch 15: val_loss improved from 1.35700 to 1.35445, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 603us/sample - loss: 1.3791 - val_loss: 1.3544\n",
      "Epoch 16/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3808\n",
      "Epoch 16: val_loss improved from 1.35445 to 1.35312, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 20s 617us/sample - loss: 1.3808 - val_loss: 1.3531\n",
      "Epoch 17/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3775\n",
      "Epoch 17: val_loss did not improve from 1.35312\n",
      "31685/31685 [==============================] - 22s 685us/sample - loss: 1.3775 - val_loss: 1.3540\n",
      "Epoch 18/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3759\n",
      "Epoch 18: val_loss improved from 1.35312 to 1.35289, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 21s 665us/sample - loss: 1.3759 - val_loss: 1.3529\n",
      "Epoch 19/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3759\n",
      "Epoch 19: val_loss did not improve from 1.35289\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.3759 - val_loss: 1.3557\n",
      "Epoch 20/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3757\n",
      "Epoch 20: val_loss improved from 1.35289 to 1.35270, saving model to ./checkpoints/unknown_person_baseline_2_p13_20.h5\n",
      "31685/31685 [==============================] - 19s 607us/sample - loss: 1.3757 - val_loss: 1.3527\n",
      "Train on 31685 samples, validate on 3538 samples\n",
      "Epoch 1/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3760\n",
      "Epoch 1: val_loss improved from inf to 1.35258, saving model to ./checkpoints/unknown_person_baseline_p13_21.h5\n",
      "31685/31685 [==============================] - 21s 660us/sample - loss: 1.3760 - val_loss: 1.3526\n",
      "Epoch 2/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3729\n",
      "Epoch 2: val_loss improved from 1.35258 to 1.34958, saving model to ./checkpoints/unknown_person_baseline_p13_21.h5\n",
      "31685/31685 [==============================] - 21s 663us/sample - loss: 1.3729 - val_loss: 1.3496\n",
      "Epoch 3/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3709\n",
      "Epoch 3: val_loss did not improve from 1.34958\n",
      "31685/31685 [==============================] - 22s 690us/sample - loss: 1.3709 - val_loss: 1.3550\n",
      "Epoch 4/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3708\n",
      "Epoch 4: val_loss improved from 1.34958 to 1.34851, saving model to ./checkpoints/unknown_person_baseline_p13_21.h5\n",
      "31685/31685 [==============================] - 22s 698us/sample - loss: 1.3708 - val_loss: 1.3485\n",
      "Epoch 5/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3718\n",
      "Epoch 5: val_loss did not improve from 1.34851\n",
      "31685/31685 [==============================] - 22s 696us/sample - loss: 1.3718 - val_loss: 1.3632\n",
      "Epoch 6/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3681\n",
      "Epoch 6: val_loss did not improve from 1.34851\n",
      "31685/31685 [==============================] - 20s 646us/sample - loss: 1.3681 - val_loss: 1.3515\n",
      "Epoch 7/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3656\n",
      "Epoch 7: val_loss did not improve from 1.34851\n",
      "31685/31685 [==============================] - 21s 653us/sample - loss: 1.3656 - val_loss: 1.3512\n",
      "Epoch 8/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3760\n",
      "Epoch 8: val_loss did not improve from 1.34851\n",
      "31685/31685 [==============================] - 19s 607us/sample - loss: 1.3760 - val_loss: 1.3519\n",
      "Epoch 9/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4038\n",
      "Epoch 9: val_loss did not improve from 1.34851\n",
      "31685/31685 [==============================] - 20s 635us/sample - loss: 1.4038 - val_loss: 1.3520\n",
      "Epoch 10/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3758\n",
      "Epoch 10: val_loss did not improve from 1.34851\n",
      "31685/31685 [==============================] - 22s 688us/sample - loss: 1.3758 - val_loss: 1.3527\n",
      "Epoch 11/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3787\n",
      "Epoch 11: val_loss did not improve from 1.34851\n",
      "31685/31685 [==============================] - 22s 683us/sample - loss: 1.3787 - val_loss: 1.3539\n",
      "Epoch 12/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3887\n",
      "Epoch 12: val_loss improved from 1.34851 to 1.34542, saving model to ./checkpoints/unknown_person_baseline_p13_21.h5\n",
      "31685/31685 [==============================] - 20s 626us/sample - loss: 1.3887 - val_loss: 1.3454\n",
      "Epoch 13/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3935\n",
      "Epoch 13: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 682us/sample - loss: 1.3935 - val_loss: 1.3559\n",
      "Epoch 14/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3962\n",
      "Epoch 14: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 21s 665us/sample - loss: 1.3962 - val_loss: 1.3569\n",
      "Epoch 15/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3909\n",
      "Epoch 15: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 603us/sample - loss: 1.3909 - val_loss: 1.3584\n",
      "Epoch 16/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4616\n",
      "Epoch 16: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 602us/sample - loss: 1.4616 - val_loss: 1.3691\n",
      "Epoch 17/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4242\n",
      "Epoch 17: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 607us/sample - loss: 1.4242 - val_loss: 1.3683\n",
      "Epoch 18/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3994\n",
      "Epoch 18: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 608us/sample - loss: 1.3994 - val_loss: 1.3621\n",
      "Epoch 19/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4095\n",
      "Epoch 19: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 606us/sample - loss: 1.4095 - val_loss: 1.3667\n",
      "Epoch 20/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4135\n",
      "Epoch 20: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 613us/sample - loss: 1.4135 - val_loss: 1.3681\n",
      "Epoch 21/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4253\n",
      "Epoch 21: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.4253 - val_loss: 1.3785\n",
      "Epoch 22/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.5657\n",
      "Epoch 22: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 20s 631us/sample - loss: 1.5657 - val_loss: 1.4109\n",
      "Epoch 23/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4644\n",
      "Epoch 23: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 692us/sample - loss: 1.4644 - val_loss: 1.3898\n",
      "Epoch 24/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4353\n",
      "Epoch 24: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 21s 661us/sample - loss: 1.4353 - val_loss: 1.3880\n",
      "Epoch 25/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4215\n",
      "Epoch 25: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 20s 625us/sample - loss: 1.4215 - val_loss: 1.3838\n",
      "Epoch 26/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4128\n",
      "Epoch 26: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 21s 672us/sample - loss: 1.4128 - val_loss: 1.3823\n",
      "Epoch 27/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4113\n",
      "Epoch 27: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 682us/sample - loss: 1.4113 - val_loss: 1.3718\n",
      "Epoch 28/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4054\n",
      "Epoch 28: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 691us/sample - loss: 1.4054 - val_loss: 1.3715\n",
      "Epoch 29/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4045\n",
      "Epoch 29: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 687us/sample - loss: 1.4045 - val_loss: 1.3700\n",
      "Epoch 30/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4013\n",
      "Epoch 30: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 21s 659us/sample - loss: 1.4013 - val_loss: 1.3670\n",
      "Epoch 31/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3979\n",
      "Epoch 31: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 683us/sample - loss: 1.3979 - val_loss: 1.3691\n",
      "Epoch 32/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3978\n",
      "Epoch 32: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 695us/sample - loss: 1.3978 - val_loss: 1.3709\n",
      "Epoch 33/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3981\n",
      "Epoch 33: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 690us/sample - loss: 1.3981 - val_loss: 1.3660\n",
      "Epoch 34/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3944\n",
      "Epoch 34: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 692us/sample - loss: 1.3944 - val_loss: 1.3689\n",
      "Epoch 35/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3916\n",
      "Epoch 35: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 690us/sample - loss: 1.3916 - val_loss: 1.3641\n",
      "Epoch 36/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3916\n",
      "Epoch 36: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 20s 618us/sample - loss: 1.3916 - val_loss: 1.3664\n",
      "Epoch 37/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3896\n",
      "Epoch 37: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 20s 616us/sample - loss: 1.3896 - val_loss: 1.3625\n",
      "Epoch 38/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3881\n",
      "Epoch 38: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 606us/sample - loss: 1.3881 - val_loss: 1.3619\n",
      "Epoch 39/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3864\n",
      "Epoch 39: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 20s 637us/sample - loss: 1.3864 - val_loss: 1.3655\n",
      "Epoch 40/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3842\n",
      "Epoch 40: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 20s 630us/sample - loss: 1.3842 - val_loss: 1.3584\n",
      "Epoch 41/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3840\n",
      "Epoch 41: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 687us/sample - loss: 1.3840 - val_loss: 1.3615\n",
      "Epoch 42/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3839\n",
      "Epoch 42: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 682us/sample - loss: 1.3839 - val_loss: 1.3589\n",
      "Epoch 43/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3798\n",
      "Epoch 43: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 681us/sample - loss: 1.3798 - val_loss: 1.3548\n",
      "Epoch 44/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3821\n",
      "Epoch 44: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 685us/sample - loss: 1.3821 - val_loss: 1.3557\n",
      "Epoch 45/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3804\n",
      "Epoch 45: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 686us/sample - loss: 1.3804 - val_loss: 1.3630\n",
      "Epoch 46/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3800\n",
      "Epoch 46: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 685us/sample - loss: 1.3800 - val_loss: 1.3541\n",
      "Epoch 47/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3764\n",
      "Epoch 47: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 22s 683us/sample - loss: 1.3764 - val_loss: 1.3522\n",
      "Epoch 48/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3743\n",
      "Epoch 48: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 20s 620us/sample - loss: 1.3743 - val_loss: 1.3580\n",
      "Epoch 49/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3748\n",
      "Epoch 49: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 599us/sample - loss: 1.3748 - val_loss: 1.3543\n",
      "Epoch 50/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3741\n",
      "Epoch 50: val_loss did not improve from 1.34542\n",
      "31685/31685 [==============================] - 19s 600us/sample - loss: 1.3741 - val_loss: 1.3519\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:19:49.009302: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_36_3/lstm_cell_147/recurrent_kernel/Assign' id:63180 op device:{requested: '', assigned: ''} def:{{{node lstm_36_3/lstm_cell_147/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_36_3/lstm_cell_147/recurrent_kernel, lstm_36_3/lstm_cell_147/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:19:52.207567: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_1_3/lstm_cell_112/recurrent_kernel/m/Assign' id:64030 op device:{requested: '', assigned: ''} def:{{{node lstm_1_3/lstm_cell_112/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_1_3/lstm_cell_112/recurrent_kernel/m, lstm_1_3/lstm_cell_112/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:19:55.259007: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1778)\n",
      "(1514, 1778)\n",
      "(1644, 1778)\n",
      "(1764, 1778)\n",
      "(1836, 1778)\n",
      "(1699, 1778)\n",
      "(1369, 1778)\n",
      "(1778, 1778)\n",
      "(1631, 1778)\n",
      "(1692, 1778)\n",
      "(1920, 1778)\n",
      "(1679, 1778)\n",
      "(1800, 1778)\n",
      "(1860, 1778)\n",
      "(1692, 1778)\n",
      "(1824, 1778)\n",
      "(958, 1778)\n",
      "(1680, 1778)\n",
      "(1860, 1778)\n",
      "{1: 8.498258547684358, 2: 5.969610363222674, 4: 7.586858887732394, 5: 7.597848151491975, 6: 5.600748335700635, 8: 8.571016176242258, 9: 4.982006590107938, 10: 7.629029444632934, 11: 5.439352030224788, 12: 8.989995993797095, 17: 8.246981450705494, 19: 7.092140749984186, 21: 10.0, 22: 1.0, 25: 7.329137362921729, 26: 7.250172995125467, 27: 2.502498605496357, 28: 5.780655598812324, 29: 2.67474232831312}\n",
      "Train on 31685 samples, validate on 3538 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:23:54.219721: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31685/31685 [==============================] - ETA: 0s - loss: 9.5309\n",
      "Epoch 1: val_loss improved from inf to 1.35589, saving model to ./checkpoints/unknown_person_p13_21.h5\n",
      "31685/31685 [==============================] - 31s 983us/sample - loss: 9.5309 - val_loss: 1.3559\n",
      "Epoch 2/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.4363\n",
      "Epoch 2: val_loss did not improve from 1.35589\n",
      "31685/31685 [==============================] - 22s 695us/sample - loss: 9.4363 - val_loss: 1.3685\n",
      "Epoch 3/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.4529\n",
      "Epoch 3: val_loss did not improve from 1.35589\n",
      "31685/31685 [==============================] - 21s 651us/sample - loss: 9.4529 - val_loss: 1.3590\n",
      "Epoch 4/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.3708\n",
      "Epoch 4: val_loss improved from 1.35589 to 1.35152, saving model to ./checkpoints/unknown_person_p13_21.h5\n",
      "31685/31685 [==============================] - 19s 608us/sample - loss: 9.3708 - val_loss: 1.3515\n",
      "Epoch 5/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.3211\n",
      "Epoch 5: val_loss improved from 1.35152 to 1.34948, saving model to ./checkpoints/unknown_person_p13_21.h5\n",
      "31685/31685 [==============================] - 21s 663us/sample - loss: 9.3211 - val_loss: 1.3495\n",
      "Epoch 6/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2998\n",
      "Epoch 6: val_loss improved from 1.34948 to 1.34864, saving model to ./checkpoints/unknown_person_p13_21.h5\n",
      "31685/31685 [==============================] - 22s 694us/sample - loss: 9.2998 - val_loss: 1.3486\n",
      "Epoch 7/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2960\n",
      "Epoch 7: val_loss did not improve from 1.34864\n",
      "31685/31685 [==============================] - 22s 690us/sample - loss: 9.2960 - val_loss: 1.3622\n",
      "Epoch 8/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2716\n",
      "Epoch 8: val_loss improved from 1.34864 to 1.34787, saving model to ./checkpoints/unknown_person_p13_21.h5\n",
      "31685/31685 [==============================] - 22s 695us/sample - loss: 9.2716 - val_loss: 1.3479\n",
      "Epoch 9/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2528\n",
      "Epoch 9: val_loss improved from 1.34787 to 1.34465, saving model to ./checkpoints/unknown_person_p13_21.h5\n",
      "31685/31685 [==============================] - 22s 691us/sample - loss: 9.2528 - val_loss: 1.3447\n",
      "Epoch 10/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2147\n",
      "Epoch 10: val_loss did not improve from 1.34465\n",
      "31685/31685 [==============================] - 22s 686us/sample - loss: 9.2147 - val_loss: 1.3495\n",
      "Epoch 11/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2087\n",
      "Epoch 11: val_loss improved from 1.34465 to 1.34440, saving model to ./checkpoints/unknown_person_p13_21.h5\n",
      "31685/31685 [==============================] - 22s 694us/sample - loss: 9.2087 - val_loss: 1.3444\n",
      "Epoch 12/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2150\n",
      "Epoch 12: val_loss did not improve from 1.34440\n",
      "31685/31685 [==============================] - 22s 691us/sample - loss: 9.2150 - val_loss: 1.3457\n",
      "Epoch 13/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2016\n",
      "Epoch 13: val_loss did not improve from 1.34440\n",
      "31685/31685 [==============================] - 21s 673us/sample - loss: 9.2016 - val_loss: 1.3497\n",
      "Epoch 14/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1762\n",
      "Epoch 14: val_loss did not improve from 1.34440\n",
      "31685/31685 [==============================] - 19s 609us/sample - loss: 9.1762 - val_loss: 1.3481\n",
      "Epoch 15/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1748\n",
      "Epoch 15: val_loss did not improve from 1.34440\n",
      "31685/31685 [==============================] - 20s 638us/sample - loss: 9.1748 - val_loss: 1.3482\n",
      "Epoch 16/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1559\n",
      "Epoch 16: val_loss did not improve from 1.34440\n",
      "31685/31685 [==============================] - 20s 621us/sample - loss: 9.1559 - val_loss: 1.3572\n",
      "Epoch 17/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1577\n",
      "Epoch 17: val_loss did not improve from 1.34440\n",
      "31685/31685 [==============================] - 20s 641us/sample - loss: 9.1577 - val_loss: 1.3538\n",
      "Epoch 18/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1488\n",
      "Epoch 18: val_loss did not improve from 1.34440\n",
      "31685/31685 [==============================] - 19s 605us/sample - loss: 9.1488 - val_loss: 1.3519\n",
      "Epoch 19/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1250\n",
      "Epoch 19: val_loss improved from 1.34440 to 1.34298, saving model to ./checkpoints/unknown_person_p13_21.h5\n",
      "31685/31685 [==============================] - 20s 628us/sample - loss: 9.1250 - val_loss: 1.3430\n",
      "Epoch 20/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1231\n",
      "Epoch 20: val_loss did not improve from 1.34298\n",
      "31685/31685 [==============================] - 21s 678us/sample - loss: 9.1231 - val_loss: 1.3450\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:31:05.363567: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_7/bias/Assign' id:82723 op device:{requested: '', assigned: ''} def:{{{node dense_7/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_7/bias, dense_7/bias/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:31:09.581158: W tensorflow/c/c_api.cc:304] Operation '{name:'conv2d_2_4/bias/m/Assign' id:83393 op device:{requested: '', assigned: ''} def:{{{node conv2d_2_4/bias/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](conv2d_2_4/bias/m, conv2d_2_4/bias/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31685 samples, validate on 3538 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:31:16.270913: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:31:38.432739: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3943"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:31:59.471832: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.35142, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 30s 962us/sample - loss: 1.3943 - val_loss: 1.3514\n",
      "Epoch 2/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3762\n",
      "Epoch 2: val_loss improved from 1.35142 to 1.34948, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 20s 615us/sample - loss: 1.3762 - val_loss: 1.3495\n",
      "Epoch 3/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3817\n",
      "Epoch 3: val_loss improved from 1.34948 to 1.34873, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 21s 650us/sample - loss: 1.3817 - val_loss: 1.3487\n",
      "Epoch 4/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3733\n",
      "Epoch 4: val_loss did not improve from 1.34873\n",
      "31685/31685 [==============================] - 22s 687us/sample - loss: 1.3733 - val_loss: 1.3493\n",
      "Epoch 5/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3730\n",
      "Epoch 5: val_loss did not improve from 1.34873\n",
      "31685/31685 [==============================] - 22s 684us/sample - loss: 1.3730 - val_loss: 1.3497\n",
      "Epoch 6/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3687\n",
      "Epoch 6: val_loss improved from 1.34873 to 1.34697, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 21s 677us/sample - loss: 1.3687 - val_loss: 1.3470\n",
      "Epoch 7/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3666\n",
      "Epoch 7: val_loss improved from 1.34697 to 1.34544, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 19s 613us/sample - loss: 1.3666 - val_loss: 1.3454\n",
      "Epoch 8/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3709\n",
      "Epoch 8: val_loss improved from 1.34544 to 1.34338, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 19s 605us/sample - loss: 1.3709 - val_loss: 1.3434\n",
      "Epoch 9/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3649\n",
      "Epoch 9: val_loss improved from 1.34338 to 1.34198, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 19s 605us/sample - loss: 1.3649 - val_loss: 1.3420\n",
      "Epoch 10/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3650\n",
      "Epoch 10: val_loss did not improve from 1.34198\n",
      "31685/31685 [==============================] - 19s 599us/sample - loss: 1.3650 - val_loss: 1.3433\n",
      "Epoch 11/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3621\n",
      "Epoch 11: val_loss did not improve from 1.34198\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.3621 - val_loss: 1.3442\n",
      "Epoch 12/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3610\n",
      "Epoch 12: val_loss did not improve from 1.34198\n",
      "31685/31685 [==============================] - 19s 608us/sample - loss: 1.3610 - val_loss: 1.3422\n",
      "Epoch 13/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3611\n",
      "Epoch 13: val_loss improved from 1.34198 to 1.33880, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 20s 618us/sample - loss: 1.3611 - val_loss: 1.3388\n",
      "Epoch 14/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3615\n",
      "Epoch 14: val_loss did not improve from 1.33880\n",
      "31685/31685 [==============================] - 19s 601us/sample - loss: 1.3615 - val_loss: 1.3414\n",
      "Epoch 15/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3568\n",
      "Epoch 15: val_loss did not improve from 1.33880\n",
      "31685/31685 [==============================] - 22s 685us/sample - loss: 1.3568 - val_loss: 1.3457\n",
      "Epoch 16/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3585\n",
      "Epoch 16: val_loss did not improve from 1.33880\n",
      "31685/31685 [==============================] - 22s 693us/sample - loss: 1.3585 - val_loss: 1.3395\n",
      "Epoch 17/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3599\n",
      "Epoch 17: val_loss did not improve from 1.33880\n",
      "31685/31685 [==============================] - 22s 691us/sample - loss: 1.3599 - val_loss: 1.3413\n",
      "Epoch 18/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3567\n",
      "Epoch 18: val_loss did not improve from 1.33880\n",
      "31685/31685 [==============================] - 22s 685us/sample - loss: 1.3567 - val_loss: 1.3478\n",
      "Epoch 19/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3557\n",
      "Epoch 19: val_loss improved from 1.33880 to 1.33872, saving model to ./checkpoints/unknown_person_baseline_2_p13_21.h5\n",
      "31685/31685 [==============================] - 19s 606us/sample - loss: 1.3557 - val_loss: 1.3387\n",
      "Epoch 20/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3538\n",
      "Epoch 20: val_loss did not improve from 1.33872\n",
      "31685/31685 [==============================] - 20s 638us/sample - loss: 1.3538 - val_loss: 1.3407\n",
      "Train on 31685 samples, validate on 3538 samples\n",
      "Epoch 1/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3541\n",
      "Epoch 1: val_loss improved from inf to 1.34016, saving model to ./checkpoints/unknown_person_baseline_p13_22.h5\n",
      "31685/31685 [==============================] - 21s 676us/sample - loss: 1.3541 - val_loss: 1.3402\n",
      "Epoch 2/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3527\n",
      "Epoch 2: val_loss improved from 1.34016 to 1.33983, saving model to ./checkpoints/unknown_person_baseline_p13_22.h5\n",
      "31685/31685 [==============================] - 21s 658us/sample - loss: 1.3527 - val_loss: 1.3398\n",
      "Epoch 3/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3489\n",
      "Epoch 3: val_loss improved from 1.33983 to 1.33806, saving model to ./checkpoints/unknown_person_baseline_p13_22.h5\n",
      "31685/31685 [==============================] - 20s 619us/sample - loss: 1.3489 - val_loss: 1.3381\n",
      "Epoch 4/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3521\n",
      "Epoch 4: val_loss did not improve from 1.33806\n",
      "31685/31685 [==============================] - 21s 678us/sample - loss: 1.3521 - val_loss: 1.3441\n",
      "Epoch 5/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3489\n",
      "Epoch 5: val_loss did not improve from 1.33806\n",
      "31685/31685 [==============================] - 22s 683us/sample - loss: 1.3489 - val_loss: 1.3396\n",
      "Epoch 6/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3451\n",
      "Epoch 6: val_loss improved from 1.33806 to 1.33650, saving model to ./checkpoints/unknown_person_baseline_p13_22.h5\n",
      "31685/31685 [==============================] - 22s 687us/sample - loss: 1.3451 - val_loss: 1.3365\n",
      "Epoch 7/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3444\n",
      "Epoch 7: val_loss did not improve from 1.33650\n",
      "31685/31685 [==============================] - 22s 692us/sample - loss: 1.3444 - val_loss: 1.3381\n",
      "Epoch 8/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3471\n",
      "Epoch 8: val_loss did not improve from 1.33650\n",
      "31685/31685 [==============================] - 22s 688us/sample - loss: 1.3471 - val_loss: 1.3423\n",
      "Epoch 9/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3548\n",
      "Epoch 9: val_loss did not improve from 1.33650\n",
      "31685/31685 [==============================] - 22s 683us/sample - loss: 1.3548 - val_loss: 1.3411\n",
      "Epoch 10/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3570\n",
      "Epoch 10: val_loss improved from 1.33650 to 1.33544, saving model to ./checkpoints/unknown_person_baseline_p13_22.h5\n",
      "31685/31685 [==============================] - 21s 665us/sample - loss: 1.3570 - val_loss: 1.3354\n",
      "Epoch 11/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3585\n",
      "Epoch 11: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 595us/sample - loss: 1.3585 - val_loss: 1.3462\n",
      "Epoch 12/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4126\n",
      "Epoch 12: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 596us/sample - loss: 1.4126 - val_loss: 1.3546\n",
      "Epoch 13/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4050\n",
      "Epoch 13: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 611us/sample - loss: 1.4050 - val_loss: 1.3540\n",
      "Epoch 14/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3904\n",
      "Epoch 14: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 20s 622us/sample - loss: 1.3904 - val_loss: 1.3467\n",
      "Epoch 15/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3803\n",
      "Epoch 15: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 607us/sample - loss: 1.3803 - val_loss: 1.3478\n",
      "Epoch 16/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3980\n",
      "Epoch 16: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 593us/sample - loss: 1.3980 - val_loss: 1.3568\n",
      "Epoch 17/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3949\n",
      "Epoch 17: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 593us/sample - loss: 1.3949 - val_loss: 1.3657\n",
      "Epoch 18/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4980\n",
      "Epoch 18: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 20s 630us/sample - loss: 1.4980 - val_loss: 1.3672\n",
      "Epoch 19/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4203\n",
      "Epoch 19: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 20s 626us/sample - loss: 1.4203 - val_loss: 1.3679\n",
      "Epoch 20/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4357\n",
      "Epoch 20: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 598us/sample - loss: 1.4357 - val_loss: 1.3706\n",
      "Epoch 21/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4764\n",
      "Epoch 21: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 594us/sample - loss: 1.4764 - val_loss: 1.3817\n",
      "Epoch 22/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.5188\n",
      "Epoch 22: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 593us/sample - loss: 1.5188 - val_loss: 1.3929\n",
      "Epoch 23/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.5232\n",
      "Epoch 23: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 601us/sample - loss: 1.5232 - val_loss: 1.3936\n",
      "Epoch 24/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4664\n",
      "Epoch 24: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 21s 660us/sample - loss: 1.4664 - val_loss: 1.3840\n",
      "Epoch 25/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4529\n",
      "Epoch 25: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 22s 685us/sample - loss: 1.4529 - val_loss: 1.3834\n",
      "Epoch 26/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4329\n",
      "Epoch 26: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 20s 628us/sample - loss: 1.4329 - val_loss: 1.3835\n",
      "Epoch 27/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4240\n",
      "Epoch 27: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 21s 659us/sample - loss: 1.4240 - val_loss: 1.3829\n",
      "Epoch 28/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4193\n",
      "Epoch 28: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 603us/sample - loss: 1.4193 - val_loss: 1.3797\n",
      "Epoch 29/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4146\n",
      "Epoch 29: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 599us/sample - loss: 1.4146 - val_loss: 1.3783\n",
      "Epoch 30/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4188\n",
      "Epoch 30: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 607us/sample - loss: 1.4188 - val_loss: 1.3770\n",
      "Epoch 31/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4118\n",
      "Epoch 31: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 22s 688us/sample - loss: 1.4118 - val_loss: 1.3750\n",
      "Epoch 32/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4147\n",
      "Epoch 32: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 22s 687us/sample - loss: 1.4147 - val_loss: 1.3782\n",
      "Epoch 33/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4102\n",
      "Epoch 33: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 605us/sample - loss: 1.4102 - val_loss: 1.3763\n",
      "Epoch 34/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4073\n",
      "Epoch 34: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 609us/sample - loss: 1.4073 - val_loss: 1.3730\n",
      "Epoch 35/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4127\n",
      "Epoch 35: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 22s 683us/sample - loss: 1.4127 - val_loss: 1.3752\n",
      "Epoch 36/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4045\n",
      "Epoch 36: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 21s 650us/sample - loss: 1.4045 - val_loss: 1.3717\n",
      "Epoch 37/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4019\n",
      "Epoch 37: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 599us/sample - loss: 1.4019 - val_loss: 1.3690\n",
      "Epoch 38/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4005\n",
      "Epoch 38: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 20s 619us/sample - loss: 1.4005 - val_loss: 1.3731\n",
      "Epoch 39/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4054\n",
      "Epoch 39: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.4054 - val_loss: 1.3741\n",
      "Epoch 40/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4138\n",
      "Epoch 40: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 599us/sample - loss: 1.4138 - val_loss: 1.3762\n",
      "Epoch 41/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4043\n",
      "Epoch 41: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 598us/sample - loss: 1.4043 - val_loss: 1.3743\n",
      "Epoch 42/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4022\n",
      "Epoch 42: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 603us/sample - loss: 1.4022 - val_loss: 1.3713\n",
      "Epoch 43/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.4033\n",
      "Epoch 43: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.4033 - val_loss: 1.3699\n",
      "Epoch 44/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3986\n",
      "Epoch 44: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 604us/sample - loss: 1.3986 - val_loss: 1.3668\n",
      "Epoch 45/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3974\n",
      "Epoch 45: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 19s 598us/sample - loss: 1.3974 - val_loss: 1.3702\n",
      "Epoch 46/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3966\n",
      "Epoch 46: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 22s 683us/sample - loss: 1.3966 - val_loss: 1.3699\n",
      "Epoch 47/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3915\n",
      "Epoch 47: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 22s 690us/sample - loss: 1.3915 - val_loss: 1.3649\n",
      "Epoch 48/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3939\n",
      "Epoch 48: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 22s 686us/sample - loss: 1.3939 - val_loss: 1.3636\n",
      "Epoch 49/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3915\n",
      "Epoch 49: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 21s 651us/sample - loss: 1.3915 - val_loss: 1.3698\n",
      "Epoch 50/50\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3857\n",
      "Epoch 50: val_loss did not improve from 1.33544\n",
      "31685/31685 [==============================] - 21s 666us/sample - loss: 1.3857 - val_loss: 1.3617\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:55:23.504601: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_30_5/lstm_cell_215/recurrent_kernel/Assign' id:100978 op device:{requested: '', assigned: ''} def:{{{node lstm_30_5/lstm_cell_215/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_30_5/lstm_cell_215/recurrent_kernel, lstm_30_5/lstm_cell_215/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:55:28.578063: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_12_5/lstm_cell_197/recurrent_kernel/m/Assign' id:102953 op device:{requested: '', assigned: ''} def:{{{node lstm_12_5/lstm_cell_197/recurrent_kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_12_5/lstm_cell_197/recurrent_kernel/m, lstm_12_5/lstm_cell_197/recurrent_kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 19:55:33.464048: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1778)\n",
      "(1514, 1778)\n",
      "(1644, 1778)\n",
      "(1764, 1778)\n",
      "(1836, 1778)\n",
      "(1699, 1778)\n",
      "(1369, 1778)\n",
      "(1778, 1778)\n",
      "(1631, 1778)\n",
      "(1692, 1778)\n",
      "(1920, 1778)\n",
      "(1679, 1778)\n",
      "(1800, 1778)\n",
      "(1860, 1778)\n",
      "(1692, 1778)\n",
      "(1824, 1778)\n",
      "(958, 1778)\n",
      "(1680, 1778)\n",
      "(1860, 1778)\n",
      "{1: 8.469559171434886, 2: 6.189419383581361, 4: 7.7760995736738705, 5: 7.949847151984789, 6: 5.5296617672756545, 8: 8.864511756186472, 9: 5.5782755416157315, 10: 7.602075591881513, 11: 5.569680792854251, 12: 8.972749960007022, 17: 8.323451941267955, 19: 7.182792503206727, 21: 10.0, 22: 1.0, 25: 7.398063581742053, 26: 6.977389728714454, 27: 2.7985154683901907, 28: 5.745641348405811, 29: 2.796643622028541}\n",
      "Train on 31685 samples, validate on 3538 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 19:59:53.246876: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31685/31685 [==============================] - ETA: 0s - loss: 9.5842\n",
      "Epoch 1: val_loss improved from inf to 1.37167, saving model to ./checkpoints/unknown_person_p13_22.h5\n",
      "31685/31685 [==============================] - 34s 1ms/sample - loss: 9.5842 - val_loss: 1.3717\n",
      "Epoch 2/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.4473\n",
      "Epoch 2: val_loss improved from 1.37167 to 1.35306, saving model to ./checkpoints/unknown_person_p13_22.h5\n",
      "31685/31685 [==============================] - 22s 693us/sample - loss: 9.4473 - val_loss: 1.3531\n",
      "Epoch 3/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.6339\n",
      "Epoch 3: val_loss did not improve from 1.35306\n",
      "31685/31685 [==============================] - 22s 686us/sample - loss: 9.6339 - val_loss: 1.3551\n",
      "Epoch 4/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.3929\n",
      "Epoch 4: val_loss did not improve from 1.35306\n",
      "31685/31685 [==============================] - 21s 663us/sample - loss: 9.3929 - val_loss: 1.3570\n",
      "Epoch 5/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.3667\n",
      "Epoch 5: val_loss improved from 1.35306 to 1.33708, saving model to ./checkpoints/unknown_person_p13_22.h5\n",
      "31685/31685 [==============================] - 21s 672us/sample - loss: 9.3667 - val_loss: 1.3371\n",
      "Epoch 6/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.3131\n",
      "Epoch 6: val_loss did not improve from 1.33708\n",
      "31685/31685 [==============================] - 20s 619us/sample - loss: 9.3131 - val_loss: 1.3402\n",
      "Epoch 7/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.3107\n",
      "Epoch 7: val_loss did not improve from 1.33708\n",
      "31685/31685 [==============================] - 22s 684us/sample - loss: 9.3107 - val_loss: 1.3448\n",
      "Epoch 8/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.3223\n",
      "Epoch 8: val_loss did not improve from 1.33708\n",
      "31685/31685 [==============================] - 21s 675us/sample - loss: 9.3223 - val_loss: 1.3398\n",
      "Epoch 9/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2881\n",
      "Epoch 9: val_loss did not improve from 1.33708\n",
      "31685/31685 [==============================] - 19s 601us/sample - loss: 9.2881 - val_loss: 1.3491\n",
      "Epoch 10/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2840\n",
      "Epoch 10: val_loss improved from 1.33708 to 1.33610, saving model to ./checkpoints/unknown_person_p13_22.h5\n",
      "31685/31685 [==============================] - 19s 607us/sample - loss: 9.2840 - val_loss: 1.3361\n",
      "Epoch 11/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2440\n",
      "Epoch 11: val_loss did not improve from 1.33610\n",
      "31685/31685 [==============================] - 21s 655us/sample - loss: 9.2440 - val_loss: 1.3386\n",
      "Epoch 12/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2397\n",
      "Epoch 12: val_loss did not improve from 1.33610\n",
      "31685/31685 [==============================] - 19s 605us/sample - loss: 9.2397 - val_loss: 1.3387\n",
      "Epoch 13/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2476\n",
      "Epoch 13: val_loss did not improve from 1.33610\n",
      "31685/31685 [==============================] - 19s 598us/sample - loss: 9.2476 - val_loss: 1.3393\n",
      "Epoch 14/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2436\n",
      "Epoch 14: val_loss did not improve from 1.33610\n",
      "31685/31685 [==============================] - 19s 602us/sample - loss: 9.2436 - val_loss: 1.3485\n",
      "Epoch 15/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2181\n",
      "Epoch 15: val_loss did not improve from 1.33610\n",
      "31685/31685 [==============================] - 21s 652us/sample - loss: 9.2181 - val_loss: 1.3454\n",
      "Epoch 16/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2287\n",
      "Epoch 16: val_loss did not improve from 1.33610\n",
      "31685/31685 [==============================] - 21s 665us/sample - loss: 9.2287 - val_loss: 1.3372\n",
      "Epoch 17/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.2179\n",
      "Epoch 17: val_loss did not improve from 1.33610\n",
      "31685/31685 [==============================] - 22s 684us/sample - loss: 9.2179 - val_loss: 1.3385\n",
      "Epoch 18/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1964\n",
      "Epoch 18: val_loss did not improve from 1.33610\n",
      "31685/31685 [==============================] - 22s 684us/sample - loss: 9.1964 - val_loss: 1.3364\n",
      "Epoch 19/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1602\n",
      "Epoch 19: val_loss improved from 1.33610 to 1.33518, saving model to ./checkpoints/unknown_person_p13_22.h5\n",
      "31685/31685 [==============================] - 21s 665us/sample - loss: 9.1602 - val_loss: 1.3352\n",
      "Epoch 20/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 9.1703\n",
      "Epoch 20: val_loss did not improve from 1.33518\n",
      "31685/31685 [==============================] - 21s 660us/sample - loss: 9.1703 - val_loss: 1.3423\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:07:02.335005: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_33_6/lstm_cell_255/kernel/Assign' id:120836 op device:{requested: '', assigned: ''} def:{{{node lstm_33_6/lstm_cell_255/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_33_6/lstm_cell_255/kernel, lstm_33_6/lstm_cell_255/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 20:07:08.554939: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_19_6/lstm_cell_241/recurrent_kernel/v/Assign' id:123099 op device:{requested: '', assigned: ''} def:{{{node lstm_19_6/lstm_cell_241/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_19_6/lstm_cell_241/recurrent_kernel/v, lstm_19_6/lstm_cell_241/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31685 samples, validate on 3538 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:07:17.860602: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:07:50.535317: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3602"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 20:08:13.106995: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.34072, saving model to ./checkpoints/unknown_person_baseline_2_p13_22.h5\n",
      "31685/31685 [==============================] - 35s 1ms/sample - loss: 1.3602 - val_loss: 1.3407\n",
      "Epoch 2/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3508\n",
      "Epoch 2: val_loss did not improve from 1.34072\n",
      "31685/31685 [==============================] - 21s 674us/sample - loss: 1.3508 - val_loss: 1.3441\n",
      "Epoch 3/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3717\n",
      "Epoch 3: val_loss did not improve from 1.34072\n",
      "31685/31685 [==============================] - 22s 682us/sample - loss: 1.3717 - val_loss: 1.3456\n",
      "Epoch 4/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3618\n",
      "Epoch 4: val_loss improved from 1.34072 to 1.33894, saving model to ./checkpoints/unknown_person_baseline_2_p13_22.h5\n",
      "31685/31685 [==============================] - 22s 687us/sample - loss: 1.3618 - val_loss: 1.3389\n",
      "Epoch 5/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3545\n",
      "Epoch 5: val_loss did not improve from 1.33894\n",
      "31685/31685 [==============================] - 20s 646us/sample - loss: 1.3545 - val_loss: 1.3434\n",
      "Epoch 6/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3524\n",
      "Epoch 6: val_loss did not improve from 1.33894\n",
      "31685/31685 [==============================] - 19s 598us/sample - loss: 1.3524 - val_loss: 1.3402\n",
      "Epoch 7/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3521\n",
      "Epoch 7: val_loss did not improve from 1.33894\n",
      "31685/31685 [==============================] - 21s 675us/sample - loss: 1.3521 - val_loss: 1.3402\n",
      "Epoch 8/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3475\n",
      "Epoch 8: val_loss did not improve from 1.33894\n",
      "31685/31685 [==============================] - 21s 663us/sample - loss: 1.3475 - val_loss: 1.3411\n",
      "Epoch 9/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3486\n",
      "Epoch 9: val_loss improved from 1.33894 to 1.33390, saving model to ./checkpoints/unknown_person_baseline_2_p13_22.h5\n",
      "31685/31685 [==============================] - 22s 686us/sample - loss: 1.3486 - val_loss: 1.3339\n",
      "Epoch 10/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3448\n",
      "Epoch 10: val_loss did not improve from 1.33390\n",
      "31685/31685 [==============================] - 22s 689us/sample - loss: 1.3448 - val_loss: 1.3413\n",
      "Epoch 11/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3426\n",
      "Epoch 11: val_loss did not improve from 1.33390\n",
      "31685/31685 [==============================] - 22s 684us/sample - loss: 1.3426 - val_loss: 1.3387\n",
      "Epoch 12/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3434\n",
      "Epoch 12: val_loss did not improve from 1.33390\n",
      "31685/31685 [==============================] - 22s 681us/sample - loss: 1.3434 - val_loss: 1.3386\n",
      "Epoch 13/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3397\n",
      "Epoch 13: val_loss did not improve from 1.33390\n",
      "31685/31685 [==============================] - 20s 629us/sample - loss: 1.3397 - val_loss: 1.3410\n",
      "Epoch 14/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3432\n",
      "Epoch 14: val_loss did not improve from 1.33390\n",
      "31685/31685 [==============================] - 19s 597us/sample - loss: 1.3432 - val_loss: 1.3373\n",
      "Epoch 15/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3404\n",
      "Epoch 15: val_loss improved from 1.33390 to 1.33099, saving model to ./checkpoints/unknown_person_baseline_2_p13_22.h5\n",
      "31685/31685 [==============================] - 21s 666us/sample - loss: 1.3404 - val_loss: 1.3310\n",
      "Epoch 16/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3413\n",
      "Epoch 16: val_loss did not improve from 1.33099\n",
      "31685/31685 [==============================] - 20s 630us/sample - loss: 1.3413 - val_loss: 1.3343\n",
      "Epoch 17/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3388\n",
      "Epoch 17: val_loss did not improve from 1.33099\n",
      "31685/31685 [==============================] - 22s 685us/sample - loss: 1.3388 - val_loss: 1.3353\n",
      "Epoch 18/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3394\n",
      "Epoch 18: val_loss did not improve from 1.33099\n",
      "31685/31685 [==============================] - 22s 681us/sample - loss: 1.3394 - val_loss: 1.3333\n",
      "Epoch 19/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3396\n",
      "Epoch 19: val_loss improved from 1.33099 to 1.32912, saving model to ./checkpoints/unknown_person_baseline_2_p13_22.h5\n",
      "31685/31685 [==============================] - 22s 687us/sample - loss: 1.3396 - val_loss: 1.3291\n",
      "Epoch 20/20\n",
      "31685/31685 [==============================] - ETA: 0s - loss: 1.3353\n",
      "Epoch 20: val_loss did not improve from 1.32912\n",
      "31685/31685 [==============================] - 22s 682us/sample - loss: 1.3353 - val_loss: 1.3346\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [13]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
